pkg/operator/ceph/cluster/osd/create.go

```diff
@@ -190,6 +190,13 @@ func (c *Cluster) startProvisioningOverPVCs(config *provisionConfig, errs *provi
 		}
 		osdProps.storeConfig.DeviceClass = volume.CrushDeviceClass
 
+		// Skip OSD prepare if deployment already exists for the PVC
+		// Also skip the encryption work part to avoid overriding the existing encryption key
+		if existingDeployments.Has(dataSource.ClaimName) {
```

Comment:
This is the same check that is seen below on line 242, right? Seems like we can delete the check on line 242 since this covers it now, just for encrypted OSDs in addition?

Reply:
Weird and good catch, my goal was to move up this check.